docs/private.rst

```diff
@@ -19,45 +19,204 @@ Notebooks
+
+The main input is a simple directory tree containing the
+FITS files of individual spectra.  Each branch off the main
+tree generates a unique dataset in the database.  It is also
+expected (although not strictly required) that each branch
+contains FITS files from a single instrument.  One is
+allowed to have sub-folders in a branch, although this is
+also not recommended.
+
+Here is an example of a directory tree
+(from the test dataset in specdb)::
+
+   ├── privateDB
+   |  ├── testDB_ztbl.fits
+   |  ├── ESI
+   |  |  ├── ESI_meta.json
+   |  |  ├── SDSSJ220758.30+125944.3_F.fits
+   |  |  ├── SDSSJ220758.30+125944.3_E.fits
+   |  ├── LRIS
+   |  |  ├── LRIS_meta.json
+   |  |  ├── SDSSJ230044.36+015541.7_r600_F.fits
+   |  |  ├── SDSSJ230044.36+015541.7_b400_F.fits
+   |  ├── COS
+   |  |  ├── COS_meta.json
+   |  |  ├── J095240.17+515250.03.fits.gz
+   |  |  ├── J095243.05+515121.15.fits.gz
+
+
+Meta parameter file
+-------------------
+
+Every dataset in `specdb` must include a meta table and there
+is a required set of columns (see :doc:`meta`) for the list
+and descriptions.
+
+The code can automatically generated the meta table from the
+data files, but it highly recommended that you guide this process
+by providing a meta parameter file in each branch of the tree.
+It must be a JSON file and it must end in _meta.json.
+
+Here is an example file::
+
+   {
+      "maxpix": 60000,
+      "meta_dict": {
+         "TELESCOPE": "HST"
+      },
+      "parse_head": {
+         "DATE-OBS": "DATE",
+         "GRATING": "OPT_ELEM",
+         "INSTR": "INSTRUME",
+         "R": True
+      }
+   }
+
+This example sets the maximum number of pixels in any given file
+within the branch (default: 10000).  The values in meta_dict are
+also set for all the files in the branch. The items in
+parse_head indicate which header keyword to use for retrieving
+the corresponding values for each individual file of the branch. These
+values may be different from file to file and this is a convenient way to get
+them directly from the headers of the FITS files. The spectral resolution (`R`) may
+be dynamically calculated within `specdb`.  That is the default and it is asserted
+with `True` in this example.
+
+Redshift table
+--------------
+
+Each unique source in your database (RA, DEC) is required to also
+have a redshift.   This must be supplied as a separate Table with
+at least the following columns:
+
+==========  ======== ============================================
+Key         Type     Description
+==========  ======== ============================================
+RA          float    RA in degrees
+DEC         float    DEC in degrees
+ZEM         float    Redshift value
+ZEM_SOURCE  str      Name of the source (e.g. SDSS, BOSS)
+==========  ======== ============================================
+
+One can generate one's own table or specify any of the public `specdb`
+databases (e.g. `igmspec`). If you generate your own, place in the top-level
```

Comment:
It was not clear to me how to tell specdb to use a public dataset for making the ztbl. Can you add an example?